specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualHardDisks.json

```diff
@@ -0,0 +1,428 @@
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualharddisks under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/virtualharddisksListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "virtualharddisksProperties": {
+      "type": "object",
+      "properties": {
+        "blockSizeBytes": {
+          "type": "integer",
+          "format": "int32"
+        },
+        "diskSizeBytes": {
+          "type": "integer",
+          "format": "int64",
+          "description": "diskSizeBytes - size of the disk in GB"
+        },
+        "dynamic": {
+          "type": "boolean",
+          "description": "Boolean for enabling dynamic sizing on the virtual hard disk"
+        },
+        "logicalSectorBytes": {
+          "type": "integer",
+          "format": "int32"
+        },
+        "physicalSectorBytes": {
+          "type": "integer",
+          "format": "int32"
+        },
+        "provisioningState": {
+          "type": "string"
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "name of the object to be used in moc"
+        },
+        "status": {
+          "type": "object",
+          "description": "MOCVirtualHardDiskStatus defines the observed state of MOCVirtualHardDisk"
+        }
+      },
+      "description": "MOCVirtualHardDiskSpec defines the desired state of MOCVirtualHardDisk"
+    },
+    "virtualharddisks": {
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "x-ms-client-flatten": true,
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/virtualharddisksProperties"
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Type"
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        },
+        "extendedLocation": {
```

Comment:
>extendedLocation [](http://example.com/codeflow?start=9&length=16)

Move to property bag

Reply:
cc @JocelynBerrendonner

Reply:
@Arkanayan , If I understand properly, we should move these properties to a common json file and do a "$ref" here.

Reply:
Doing so should *as per my understanding* not be a breaking change.